### PR TITLE
Rotate x-axis labels if the labels still overflow after wrapping

### PR DESCRIPTION
**Pull request by beets**
Some examples:
image

image

image

### CODE REVIEW

**Comment by beets** [comment]
*friendly bump*

File: server/routes/place.py [hunk 1]
```diff
@@ -49,11 +49,16 @@ def place(place_dcid=None):
     dcid = flask.request.args.get('dcid', None)
     topic = flask.request.args.get('topic', None)
     if dcid:
+        # Traffic from "explore more" in Search. Forward along all parameters,
+        # except for dcid, to the new URL format.
+        redirect_args = dict(flask.request.args)
+        redirect_args['place_dcid'] = dcid
+        del redirect_args['dcid']
+        redirect_args['topic'] = topic
         url = flask.url_for('place.place',
```
Comment (shifucun) [inline server/routes/place.py:58]: is this for the utm_medium fix?

**Review by shifucun — APPROVED** [review]
Update the PR description too to include other changes?

**Comment by beets** [comment]
I made a mistake with a follow on change without switching branches :-o

Could you please re-review? I'll send another PR with the changes you reviewed.

**Review by shifucun — APPROVED** [review]
looking good, learned a lot of d3 tricks :)

**Comment by beets** [comment]
thanks!

**Pull request merged**